Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -595,8 +597,9 @@ namespace AzToolsFramework
             
             Instance& entityOwningInstance = owningInstanceOfParentEntity->get();
 
-            PrefabDom instanceDomBeforeUpdate;
-            m_instanceToTemplateInterface->GenerateDomForInstance(instanceDomBeforeUpdate, entityOwningInstance);
+            // Get the comparison instance directly from the template, as that's what we'll be applying our patch to
+            const PrefabDom& instanceDomBeforeUpdate =
```

Comment:
Another area of note: when loading any existing level with fix-up I was getting patch application failures that spiraled into pretty bad prefab state failures without this change. I attempted to create undo stack entries every time sanitization ran and mutated the entity order component, but it was difficult to find every case in which the generated DOM and the existing template were out of sync. I'm happy to discuss this, as there may be an important reason why we need to reserialize for generating a delta that I'm missing context on.

Reply:
The idea is that a prefab instance may differ from the template due to unique overrides on it. As such, it's a good idea to get the dom of the actual instance you are editing since in theory, it can look totally different from the template it was originally created with.

Also, at the end of this function, we call PrefabUndoHelpers::UpdatePrefabInstance(), which will again compare it with the final dom of the instance itself. So, your change makes it use a hybrid approach where you use the template dom for the before state and instance dom for the after state. We should either use template dom for both or instance dom for both. Otherwise, I'm afraid we'll run into more bugs around this.

Currently, it's not possible to author overrides in the editor. So ideally there shouldn't be any difference whether you take the template dom or instance dom in most cases. So I'd be interested to see why choosing the template dom reduces errors. Does it mean that we are putting the instance in an invalid state before propagating the change to it? If so, I think we should fix that instead.

Reply:
This makes me think that the logic to extract the dom of an instance as seen as the current focus should be externalized from the propagation code and re-used in other places too. That way we know we're always accessing the correct DOM based on the state of the Editor.